pkg/cmd/server/apis/config/validation/master_test.go

```diff
@@ -525,3 +527,79 @@ func TestValidateIngressIPNetworkCIDR(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateiMasterAuthConfig(t *testing.T) {
+	testConfigFile, err := ioutil.TempFile("", "test1.cfg")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer os.Remove(testConfigFile.Name())
+
+	testCases := []struct {
+		testName                   string
+		errorCount                 int
+		RequestHeader              *configapi.RequestHeaderAuthenticationOptions
+		WebhookTokenAuthenticators []configapi.WebhookTokenAuthenticator
+	}{
+		{
+			testName: "No AuthConfig",
+		},
+		{
+			testName: "Valid RequestHeader",
+			RequestHeader: &configapi.RequestHeaderAuthenticationOptions{
+				ClientCA:            "ClientCA",
+				ClientCommonNames:   []string{"ClientCommonNames", "2", "3"},
+				UsernameHeaders:     []string{"Username", "Headers"},
+				GroupHeaders:        []string{"Group", "Headers"},
+				ExtraHeaderPrefixes: []string{"Extra", "Header", "Prefixes"},
+			},
+		},
+		{
+			testName: "Only One WebhookTokenAuthenticator",
+			WebhookTokenAuthenticators: []configapi.WebhookTokenAuthenticator{
+				{
+					ConfigFile: testConfigFile.Name(),
+					CacheTTL:   "2m",
+				},
+			},
+		},
+		{
+			testName: "Unexisting config file in second WebhookTokenAuthenticator",
+			WebhookTokenAuthenticators: []configapi.WebhookTokenAuthenticator{
+				{
+					ConfigFile: testConfigFile.Name(),
+					CacheTTL:   "2m",
+				},
+				{
+					ConfigFile: "Unexisting",
+					CacheTTL:   "2m",
+				},
+			},
+			errorCount: 1,
+		},
+		{
+			testName: "Invalid and Empty CacheTTL WebhookTokenAuthenticator",
+			WebhookTokenAuthenticators: []configapi.WebhookTokenAuthenticator{
+				{
+					ConfigFile: testConfigFile.Name(),
+					CacheTTL:   "-2m",
+				},
+				{
+					ConfigFile: testConfigFile.Name(),
+				},
+			},
+			errorCount: 2,
+		},
+	}
+	for _, test := range testCases {
+		config := configapi.MasterAuthConfig{
+			RequestHeader:              test.RequestHeader,
+			WebhookTokenAuthenticators: test.WebhookTokenAuthenticators,
+		}
+		errors := ValidateMasterAuthConfig(config, nil)
+		errorCount := len(errors.Errors)
+		if test.errorCount != errorCount {
```

Comment:
Compare the actual errors, not the count.

Reply:
done